core/chains/evm/txmgr/txmgr.go

```diff
@@ -56,7 +57,7 @@ type KeyStore interface {
 	CheckEnabled(address common.Address, chainID *big.Int) error
 	EnabledKeysForChain(chainID *big.Int) (keys []ethkey.KeyV2, err error)
 	GetNextNonce(address common.Address, chainID *big.Int, qopts ...pg.QOpt) (int64, error)
-	GetStatesForChain(chainID *big.Int) ([]ethkey.State, error)
+	GetStatesForChain(chainID *big.Int) ([]ethkey.KeyState[common.Address, *big.Int, int64], error)
 	IncrementNextNonce(address common.Address, chainID *big.Int, currentNonce int64, qopts ...pg.QOpt) error
 	SignTx(fromAddress common.Address, tx *gethTypes.Transaction, chainID *big.Int) (*gethTypes.Transaction, error)
 	SubscribeToKeyChanges() (ch chan struct{}, unsub func())
```

Comment:
generic types should be implemented in this `KeyStore` type and then the `var _ Keystore[...] = (*Eth)(nil)`

i don't think you need make major changes to the actual Eth keystore